odo/backends/pandas.py

```diff
@@ -57,9 +57,20 @@ def coerce_datetimes(df):
     name            object
     dtype: object
     """
-    df2 = df.select_dtypes(include=['object']).apply(
-        partial(pd.to_datetime, errors='ignore')
-    )
+
+    def patched_to_datetime(s):
```

Comment:
This is the same logic that we use for inferring datetime values in `datashape.discover`